Bluetooth Mesh: Scheduler Server doc (#23416)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth/mesh/scheduler_srv.rst

The Scheduler Server is implemented as a root model.

> When a Scheduler Server model is present on an element, the Scene Server model (see the :ref:`bt_mesh_scene_srv_readme` documentation) shall also be present on the same element.

Reviewer: ```suggestion
When a Scheduler Server model is present on an element, the Scene Server model (see the :ref:`bt_mesh_scene_srv_readme` documentation) or the Generic OnOff Server model (see the :ref:`bt_mesh_onoff_srv_readme` documentation) or both(depends on managed Scheduler Server actions) shall also be present on the same or lower indexed element as the Scheduler Server model.
```

Author: ONOFF server is searched on all elements:

.. note::
   In addition, the |NCS| has the following implementation-specific requirements:

   * The Time Server model (see :ref:`bt_mesh_time_srv_readme`) must be instantiated in the application.

>    * The default Transition Time Server model (see :ref:`bt_mesh_dtt_srv_readme`) must be instantiated in the application on the same or lower indexed element as the Scheduler Server model.

Reviewer: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L529-L530

>    See the :ref:`bt_mesh_scheduler_srv` API documentation for model instantiation details.

Reviewer 2: ```suggestion
   See the :ref:`bt_mesh_scheduler_srv_readme` API documentation for model instantiation details.
```
To fix the doc build failure.

The Scheduler Server is implemented as a root model.
When a Scheduler Server model is present on an element, the Scene Server model (see the :ref:`bt_mesh_scene_srv_readme` documentation) shall also be present on the same element.

.. note::

>    There are additional implementation-specific requirements in |NCS|.

Reviewer 2: ```suggestion
   In addition, the |NCS| has the following implementation-specific requirements:
```

>    Following models are required to be instantiated on the same element as the Scheduler Server model:
> 
>    * Time Server model (see the :ref:`bt_mesh_time_srv_readme` documentation)
> 
>    Following models are required to be instantiated on the device on the same or lower indexed element as the Scheduler Server model:
> 
>    * Default Transition Time Server model (see the :ref:`bt_mesh_dtt_srv_readme` documentation)
> 
>    See the :ref:`bt_mesh_scheduler_srv` API documentation for model instantiation details.

Reviewer 2: ```suggestion

   * The Time Server model (see :ref:`bt_mesh_time_srv_readme`) must be instantiated on the same element as the Scheduler Server model.
   * The default Transition Time Server model (see :ref:`bt_mesh_dtt_srv_readme`) must be instantiated on the device on the same or lower indexed element as the Scheduler Server model.

   See the :ref:`bt_mesh_scheduler_srv` API documentation for model instantiation details.
```
